src/app/leverage/provider.tsx

```diff
@@ -339,7 +339,6 @@ export function LeverageProvider(props: { children: any }) {
       if (!provider || !publicClient) return
       if (inputTokenAmount <= 0) return
       if (!indexToken) return
-      const jsonRpcProvider = publicClientToProvider(publicClient)
```

Comment:
only one instance in the project left after this 🙌 

@0xonramp check my proposal here https://github.com/IndexCoop/index-app/issues/1332 to get rid of it entirely in the near future.